### ISSUE TITLE
Bug: playground just generally doesn't work

### ISSUE THREAD

**Issue by G-Rath** [typescript-eslint/typescript-eslint#11281]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

website

### Repro Code

```TypeScript
-
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

I'm able to use the playground

### Actual Result

Most of the features don't actually seem to work - I can interact with elements like checkboxes and tabs, type in code which gets red squiggles and highlighting, but most of the interaction doesn't actually do anything e.g. switching from "code" to "tsconfig" to "eslintrc" doesn't change the editor content (though it does cause a "Visual Editor" button to appear which does nothing when clicked); likewise for the "errors", "estree", etc.

If I switch to the "ESTree" tab, I do get an "ESQuery filter" input which immediately gives me this warning when I type anything:

```
Invalid Selector
Cannot read properties of undefined (reading 'parse')
```

![Image](https://github.com/user-attachments/assets/8297b3cc-3c73-44da-b4cd-7eaeee01b58e)

I have tried this on both latest Chrome and Edge, with extensions installed like ublock origins (which isn't reporting anything being blocked, and has been disabled for the site with no luck), with no extensions, and a few other combos, all with the same result.

There doesn't seem to be any errors in the console

![Image](https://github.com/user-attachments/assets/92268769-67f4-4f33-97df-75dc4efae74d)

Here's another set of screenshots after I've clicked through some stuff and typed again into the editor:

![Image](https://github.com/user-attachments/assets/0b696766-4109-455a-8df7-4e904640b4c9)

![Image](https://github.com/user-attachments/assets/e6db555d-1080-460d-9378-f23277f8297e)

### Additional Info

I'd like to have attached a screen recording, but don't have any software for that setup currently - I can look into that next week if needed.

Also let me know if there's any sections of template which you'd like filled out in some way - I skipped most of them as they didn't seem relevant and I couldn't find a more appropriate

### Versions

-

**Comment by G-Rath** [comment]
ugh and of course having finally gotten around to opening this, I've realized the problem, though I think there's still a bug - I'm using the playground with TS v4 somehow which is no longer supported; going to https://typescript-eslint.io/play/#ts=5.8.2 fixes this.

But I still think there's something weird here as I had to go to the above link to resolve this - just removing the TS version wasn't enough, nor could I change the version of TypeScript to another; I assume it's because the version is being stored in the usual places, but at the least it might be a good idea to have a min. version check to ensure people are not out of range?

I assume this probably happened as a result of me using the playground sometime in the past when v4.3.5 was supported (causing it to be saved in my browser) and then have not used the playground until very recently.

**Comment by kirkwaiblinger** [comment]
Confirmed I can repro this. I'm thinking if an unsupported TS version is specified in the hash we should either show an error message in a useful way or just ignore the version provided and default it to latest supported 🤷‍♂
